growth-fancy/script.js

```diff
@@ -0,0 +1,243 @@
+}
+
+function getStatsFromDocument(){
+	try {
+		initDocElemVars();
+		showLoading();
+		let numberOfDays = parseFloat(document.getElementById('records').value) || 0;
+		setLoadingZoneDesc("Requesting " + numberOfDays + " days from server.");
+		createChart(numberOfDays);
+	} catch(err){
+		handleError(err);
+	}
+}
+
+function getDateStringFrom(numberOfDays) {
+	let dateStringToday = new Date().toISOString();
+	let dateStringFromTemp = dateStringToday.substring(0, 10) + "T00:00:00.000Z";
+	let dateFrom = new Date(dateStringFromTemp);
+	dateFrom.setDate(dateFrom.getDate() - numberOfDays);
+	let dateStringFrom = dateFrom.toISOString();
+	console.log(dateStringFrom);
+	return dateStringFrom;
+}
+
+function createChart(numberOfDays){
+	//Test-Data
+	/*
+	let xobj = new XMLHttpRequest();
+	xobj.overrideMimeType("application/json");
+	xobj.open("GET", "testData.json", true);
+	*/
+	
+	let request = new XMLHttpRequest();
+	request.addEventListener('load', function(event) {
+		if (request.status >= 200 && request.status < 300) {
+			displayResults(this.responseText)
+		} else {
+			handleError(request.status + " - " + request.statusText + " - " + request.responseText);
+		}
+	});
+	request.open("GET", "https://api.faforever.com/data/player?fields[player]=userAgent,createTime&filter[player]=createTime>" + getDateStringFrom(numberOfDays) /*2018-07-05T00:00Z*/ + "&page[limit]=" + recordLimit + "&page[totals]", true);
+	request.setRequestHeader("Content-type","application/vnd.api+json");
+	request.send();
+}
+
+function createSeriesObject(jsonData) {
+	let dates = {};
+	let series = {};
+	
+	for (let userIndex in jsonData.data){
+		let user = jsonData.data[userIndex];
+		let userAgent = user.attributes.userAgent;
+		let dateTime = user.attributes.createTime;
+		let date = new Date(dateTime.substring(0, 10)).getTime();
+		//create object for date to hold user agent entries if non-existant
+		dates[date] = dates[date] || {};
+		//increment user agent entry or create with 1 as value
+		dates[date][userAgent] = dates[date][userAgent] + 1 || 1;
+		
+		series[userAgent] = series[userAgent] || {};
+		series[userAgent][date] = series[userAgent][date] + 1 || 1;
+	}
+	//}
+	console.log(dates);
+	console.log(series);
+	
+	return series;
+}
+
+function createSeriesArray(seriesObject) {
+	appendToLoadingZoneDesc(" Creating chart series.");
+	//convert parsed objects to nested arrays (see testArray)
+	let seriesArray = [];
+	for (let serieIndex in seriesObject){
+		let serie = seriesObject[serieIndex];
+		
+		let serieArray = [];
+		for (let dateIndex in serie){
+			let count = serie[dateIndex];
+			
+			serieArray.push([dateIndex, count]);
+		}
+		let serieLabel = serieIndex === "null" ? "other" : serieIndex;
+		let serieObject = {label: serieLabel, data: serieArray};
+		seriesArray.push(serieObject);
+	}
+	return seriesArray;
+}
+
+function displayResults(str_result){
+	try {
+		appendToLoadingZoneDesc(" Parsing " + str_result.length + " characters to json.");
```

Comment:
The need to use spaces when calling the function is a direct consequence of what I mentioned ealier ^

Reply:
So I shall use `append`[`Child`] instead, correct?

Reply:
Create a <p> element (or another element) for your texts and then `.appendChild()` those elements to the `loadingZoneDescElem`